pkg/operator/status.go

```diff
@@ -50,29 +44,18 @@ func (optr *Operator) syncVersion() error {
 			Namespace: co.Namespace,
 			UID:       co.GetUID(),
 		}
-		optr.eventRecorder.Eventf(mcoObjectRef, corev1.EventTypeNormal, "OperatorVersionChanged", fmt.Sprintf("clusteroperator/machine-config-operator version changed from %v to %v", co.Status.Versions, optr.vStore.GetAll()))
+		optr.eventRecorder.Eventf(mcoObjectRef, corev1.EventTypeNormal, "OperatorVersionChanged", fmt.Sprintf("clusteroperator/machine-config version changed from %v to %v", co.Status.Versions, optr.vStore.GetAll()))
 	}
 
 	co.Status.Versions = optr.vStore.GetAll()
-	// TODO(runcom): abstract below with updateStatus
-	optr.setOperatorStatusExtension(&co.Status, nil)
-	_, err = optr.configClient.ConfigV1().ClusterOperators().UpdateStatus(context.TODO(), co, metav1.UpdateOptions{})
```

Comment:
Just curious, what did this used to do? It didn't seem to set much if anything. Did it just reset the status?

Reply:
By sending nil for the error in this particular call, the only update that could take potentially place is if any MCP status has changed. So we were sort of refreshing our pool status here.

Since we are batching the update status call as much as possible, I decided to do this as part of the [final update call](https://github.com/openshift/machine-config-operator/pull/4321/files/f5263e7a0fdf7279e9059e234b537cdfdacb2e26#diff-8a6a064b4db19eb3b29e5280447ac3ba9496e847391489d25c1282767c3354c5R142-R148).